src/backend/utils/adt/jsonb_util.c

```diff
@@ -676,6 +676,7 @@ appendValue(JsonbParseState *pstate, JsonbValue *scalarVal)
 {
 	JsonbValue *object = &pstate->contVal;
 
+	Insist(scalarVal);
```

Comment:
@andr-sokolov and I decided that Insist should be moved here before assert.
https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/utils/adt/jsonb_util.c#L588-L589

Reply:
Can you explain why? A couple lines below there's one more `IsAJsonbScalar` call, should we duplicate that insist?

Reply:
I think so, because the static analyzer does not see assertions and only noticed an error here.

Reply:
What if we change assert to `Assert(scalarVal && IsAJsonbScalar(scalarVal));` and keep Insist at appendValue so that the check will be at the place it should be?

Reply:
may be

Reply:
Done